conclave-enclave/src/main/kotlin/com/r3/conclave/enclave/internal/NativeEnclaveEnvironment.kt

```diff
@@ -19,33 +17,41 @@ import java.util.concurrent.atomic.AtomicLong
 
 @PotentialPackagePrivate
 class NativeEnclaveEnvironment(
-    enclaveClass: Class<*>
+    enclaveClass: Class<*>,
+    override val hostCallInterface: NativeHostCallInterface
```

Comment:
I am confused by this `hostCallInterface` member variable and the one in the companion object.
Are both of them needed?

Reply:
They are the same instance. The `EnclaveEnvironment` class is our enclave-side abstraction for dealing with different enclave implementations (the counterpart to `EnclaveHandle` on the host side). For the call interface to be agnostic to the enclave implementation, the interface instance has to be in the `EnclaveEnvironment` instance (hence the override). In the case of the `NativeEnclaveEnvironment` the call interface is a singleton (in the companion) and is passed into the `NativeEnclaveEnvironment` when the object is constructed. In mock mode it works differently, and it will probably be different still when we implement the `GrameneEnclaveEnvironment` class.